Faces dashboard scrollscrubber

5m4u66y opened the pull request:
I have been working on a scroll scrubber for faces dashboard.
There is a huge place for improvement, but I would appreciate some feedback. It might already been better than nothing.

Implements one of the suggested feature from Issue LibrePhotos/librephotos#600 

![ScrollScrubber](https://user-images.githubusercontent.com/115424459/199702648-7f01dd40-dc58-4782-8434-e935e7b0290f.gif)

5m4u66y committed:
Add a scroll scrubber to faces dashboard

5m4u66y committed:
Added a margin to bottom of scroll scrubber

5m4u66y committed:
Fix avoid overlapping letters

5m4u66y committed:
Fix linting

5m4u66y committed:
Fix type name

5m4u66y committed:
Fix ScrolScrubber size
Determine top position based on parentElement
Take into accound target visible area
Fix markers above 100% of ScrollScrubber size

5m4u66y committed:
Add "Unknown - Other" to the markers

5m4u66y committed:
Fix linting

derneuere commented:
That's a huge improvement! I have a bit of feedback on the visual side.

Visual things:

I think it would look nicer, if you would use the same colors for the dark/white theme as the rest of the page. So go for dark grey for the boxes and the lines and white for the text in dark mode. For the light mode light grey and dark grey for the text.

A nice shadow when hovering over the scroll bar would also help with visibility when there are alot of images on the right hand side.

The scrollbar clashes with the mobile navigation.

Let me know, if you want to apply these changes in this PR or if you want to split them up :)

Do you think, that you can also implement something similar for the regular PhotoListView?

5m4u66y committed:
Fix get label at Y position of scroller

5m4u66y committed:
Use theme collors

5m4u66y committed:
Fix scrollscrubber clashes with navigation bar on small screens

5m4u66y commented:
@derneuere regarding the colors, I didn't know how to use the theme colors for my css when I started to work on this. I totally agree that I need to stick with theme colors. I fixed this, hope this is better now.
I fixed the scrollbar bottom position on mobile devices. Anyway, this scroll scrubber is not very mobile friendly at the moment.
As for the "nice shadow", I am not really good with CSS and can't get something nice. So some help, improving visibility on faces would be appreciated.
If you think that "as is" this scrollscrubber is good to go in the dev branch, then you can merge this PR, I will try to improve and fix some bugs in future PR. It could help to have feedback from more users.
I would really like to do something similar for the PhotoListView. Just need to look at how to get the groups positions on the react-pig and how to scroll this component to a given position. I will look into this in the next few days.

5m4u66y committed:
Remove unused import

derneuere commented:
Looks good to me :+1:
